src/Microsoft.DocAsCode.AzureMarkdownRewriters/AzureMigrationEngineBuilder.cs

```diff
@@ -266,5 +269,31 @@ private string GenerateAzureLinkHref(IMarkdownContext context, string href, stri
 
             return azureHref;
         }
+
+        private string GenerateAzureVideoLink(IMarkdownContext context, string azureVideoId, string rawMarkdown)
+        {
+            object path;
+            if (!context.Variables.TryGetValue("path", out path))
+            {
+                Logger.LogWarning("Can't get current file path. Skip video token rewriter.");
+                return azureVideoId;
+            }
+
+            if (!context.Variables.ContainsKey("azureVideoInfoMapping"))
```

Comment:
Use that I also need to use as/cast more code is needed. This may be much more clear.

---
In reply to: [73269616](https://github.com/dotnet/docfx/pull/559#discussion_r73269616) [](ancestors = 73269616)